builder/googlecompute/config.go

```diff
@@ -632,6 +637,18 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		c.WindowsPasswordTimeout = 3 * time.Minute
 	}
 
+	if c.InstanceMaxRunDurationSeconds < 0 {
```

Comment:
My suggestion here would be to include a high default value and possibly validate the the provided value is not lower than the default timeouts set for SSH or other Packer processes. To avoid terminating the machine before establishing a connection. It might not be so straightforward to know all the timeouts so I would suggest documenting it as well. Within the configuration option that at minimum the value should be set to X number of seconds. Possibly even hours or days.


I’ll leave this question for @lbajolet-hashicorp Should this be an unsigned int?

Reply:
It seems that if it is not straightforward to know all the timeouts, setting a high default without checking the other config values would be the best path forward to me. I could set the default to something like 2 days and add documentation to the config values. I also like the idea of setting the default termination option to `STOP` so I'll add that.

> Should this be an unsigned int?
Probably. I don't believe that `maxRunDuration` can be negative.